SpatialGDK/Source/SpatialGDKFunctionalTests/SpatialGDK/EventTracingTests/EventTracingTest.cpp

```diff
@@ -155,22 +158,23 @@ void AEventTracingTest::GatherDataFromFile(const FString& FilePath)
 				const Trace_Event& Event = Item->item.event;
 				FName EventName = FName(*FString(Event.type));
 
-				FString SpanIdString = SpatialEventTracer::SpanIdToString(Event.span_id);
 				if (FilterEventNames.Num() == 0 || FilterEventNames.Contains(EventName))
 				{
+					FString SpanIdString = FSpatialGDKSpanId::ToString(Event.span_id);
 					FName& CachedEventName = TraceEvents.FindOrAdd(SpanIdString);
 					CachedEventName = EventName;
 				}
 			}
 			else
 			{
 				const Trace_Span& Span = Item->item.span;
-				FString SpanIdString = SpatialEventTracer::SpanIdToString(Span.id);
 
+				FString SpanIdString = FSpatialGDKSpanId::ToString(Span.id);
 				TArray<FString>& Causes = TraceSpans.FindOrAdd(SpanIdString);
 				for (uint64 i = 0; i < Span.cause_count; ++i)
 				{
-					Causes.Add(SpatialEventTracer::SpanIdToString(Span.causes[i]));
+					const int32 ByteOffset = i * TRACE_SPAN_ID_SIZE_BYTES * sizeof(Trace_SpanIdType);
+					Causes.Add(FSpatialGDKSpanId::ToString(Span.causes + ByteOffset));
```

Comment:
This seems wrong, I think Span.causes[i] is what you mean here

Reply:
Or Span.causes + i

Reply:
I think this is right.  

Span.causes is a pointer to uint8. 
If we access the second element of Span.causes then we wont be accessing the second SpanId cause, just a different element of the last SpanId we were processing. 

We need to jump 16 bytes from the pointer each time we read. We might be able to do Span.causes[ByteOffset] but we do need the * TRACE_SPAN_ID_SIZE_BYTES in there I think.

Reply:
Ah nice, sorry thought it was a Trace_SpanId but maybe that was one of the things that changed in the worker sdk